connman/plugins/sailfish_ofono.c

```diff
@@ -1177,7 +1202,9 @@ static void modem_create(struct plugin_data *plugin, OfonoModem *modem)
 	g_hash_table_replace(plugin->modems, g_strdup(path), md);
 
 	if (ofono_modem_valid(modem)) {
-		ofono_modem_set_powered(modem, TRUE);
+		md->pending_set_powered
+			= ofono_modem_set_powered_full(modem, TRUE,
```

Comment:
Also here we're doing a lot of unnecessary extra if the modem cannot be powered on. Online check, network create/disable, context updates, network registration. Not sure what the effect is if we have an half setup modem here with online check disabled as it cannot be powered on.

There is a comment in `modem_can_create_network()`:
```
        /*
	 * Don't create the network if cellular technology is disabled,
	 * otherwise connman will keep on trying to connect it.
	 * /
```

This is kind of weird situation as connman should not even create a device for this case and that should block from ever trying the false modem in anyway to be powered up. Granted, this callback use is good addition to error cases but I think some blacklisting might be looked at as well...